### PR TITLE
Added resource filter to market

### PR DESCRIPTION
**Pull request by jimmyh77**

### CODE REVIEW

File: src/scripts/ui/MarketBuildingBody.tsx [hunk 1]
```diff
@@ -16,6 +17,40 @@ import { BuildingStorageComponent } from "./BuildingStorageComponent";
 import { BuildingUpgradeComponent } from "./BuildingUpgradeComponent";
 import { BuildingWorkerComponent } from "./BuildingWorkerComponent";
 import { FormatNumber } from "./HelperComponents";
+import type { ResourceDefinitions } from "../../../shared/definitions/ResourceDefinitions";
+
+interface IMarketFiltersProps {
+   resourceFilterParams: {resourceFilter: string, setResourceFilter: React.Dispatch<React.SetStateAction<string>>},
```
Comment (insraq) [inline src/scripts/ui/MarketBuildingBody.tsx:23]: `React.Dispatch<React.SetStateAction<string>>` this type signature does not look right. It should be simplified.

**Review by insraq — CHANGES_REQUESTED** [review]

File: src/scripts/ui/MarketBuildingBody.tsx [hunk 2]
```diff
@@ -16,6 +17,40 @@ import { BuildingStorageComponent } from "./BuildingStorageComponent";
...
+   resources: (keyof ResourceDefinitions)[]
+}
+function MarketFilters({resourceFilterParams, resources}: IMarketFiltersProps): React.ReactNode {
```
[the remark this answers is not in the document]
Comment (jimmyh77) [reply src/scripts/ui/MarketBuildingBody.tsx:26]: I created a component so it didn't bloat the main component, but happy to move it into `MarketBuildingBody` if you want.

File: src/scripts/ui/MarketBuildingBody.tsx [hunk 3]
```diff
@@ -16,6 +17,40 @@ import { BuildingStorageComponent } from "./BuildingStorageComponent";
...
+   const { resourceFilter, setResourceFilter} = resourceFilterParams;
+
+   return (
+      <fieldset>
+         <legend>Filters</legend>
```
[the remark this answers is not in the document]
Comment (jimmyh77) [reply src/scripts/ui/MarketBuildingBody.tsx]: Added

File: src/scripts/ui/MarketBuildingBody.tsx (hunk 1 above)
Comment (jimmyh77) [reply src/scripts/ui/MarketBuildingBody.tsx]: Simplified the types.

**Comment by jimmyh77** [comment]
Happy for you to close this if it's causing an issue.

**Pull request closed without merging**